wa/workloads/schbench/__init__.py

```diff
@@ -0,0 +1,105 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+
+from wa import Workload, Parameter, Executable
+from wa.utils.exec_control import once
+
+timeout_buffer = 10
+
+regex_map = {"50.0th": re.compile(r'50.0th: (\d+)'),
+             "75.0th": re.compile(r'75.0th: (\d+)'),
+             "90.0th": re.compile(r'90.0th: (\d+)'),
+             "95.0th": re.compile(r'95.0th: (\d+)'),
+             "*99.0th": re.compile(r'\*99.0th: (\d+)'),
+             "99.5th": re.compile(r'99.5th: (\d+)'),
+             "99.9th": re.compile(r'99.9th: (\d+)'),
+             "min": re.compile(r'min=(\d+)'),
+             "max": re.compile(r'max=(\d+)')
+             }
+
+
+class Schbench(Workload):
+    name = 'schbench'
+    description = """
+    Benchmark providing detailed latency distribution statistics for scheduler
+    wakeups.
+    """
+
+    parameters = [
+        Parameter('runtime', kind=int, default=30, aliases=['timeout'],
+                  description='How long to run before exiting (seconds)'),
+        Parameter('message_threads', kind=int, default=2,
+                  description='number of message threads'),
+        Parameter('threads', kind=int, default=16,
+                  description='worker threads per message thread'),
+        Parameter('sleeptime', kind=int, default=30000,
+                  description='Message thread latency (usec)'),
+        Parameter('cputime', kind=int, default=30000,
+                  description='How long to think during loop'),
+        Parameter('auto', kind=bool, default=False,
+                  description='grow thread count until latencies hurt'),
+        Parameter('pipe', kind=int, default=0,
+                  description='transfer size bytes to simulate a pipe test'),
+        Parameter('rps', kind=int, default=0,
+                  description='requests per second mode (count)'),
+    ]
+
+    binary_name = 'schbench'
+    schbench_results_txt = 'schbench_results.txt'
+    output_unit = 'usec'
+
+    @once
+    def initialize(self, context):
+        host_binary = context.get_resource(
+            Executable(self, self.target.abi, self.binary_name))
+        Schbench.target_binary = self.target.install(host_binary)
+
+    def setup(self, context):
+        self.target_output_file = self.target.get_workpath(
+            self.schbench_results_txt)
+        self.run_timeout = self.runtime + timeout_buffer
+        self.command = f"""{self.target_binary} -m {self.message_threads} \
+            -t {self.threads} -r {self.runtime} -s {self.sleeptime} \
+            -c {self.cputime} -p {self.pipe} -R {self.rps} \
+            {'-a' if self.auto else ''}
+        """
```

Comment:
Sorry would it be possible to re-write this without using f-strings? 
I have been meaning to do a final release of WA before we drop Python 3.5 support and start using the new language features.